### ISSUE TITLE
Support for "deployment arguments"

### ISSUE THREAD

**Issue by ashur** [ashur/fig#24]
## Description

Consider adding profile support for referencing "deployment arguments," values supplied to the `deploy` command at runtime.

## Current Behavior

Assume a profile `foo/hello` which uses a variable-like pattern `{{args[n]}}`:

```yaml
# hello
---

- name: say hello
  command: echo "hello, {{ args[0] }}"
```

Currently, such a profile would be deployed with:

```
fig deploy foo/hello
```

and result in:

```
COMMAND: say hello *************************************************************
hello, {{ args[0] }}
```

because the variable `args[0]` is undefined in the profile itself.

## Desired Behavior

Passing arguments to the deployment command at runtime

```
$ fig deploy foo/hello [<args>...]
```

would make those values available to the deployed profile via a new [variable-like](https://github.com/ashur/fig/wiki/vars) pattern `{{args[n]}}`

In our example, if we pass `world` to `foo/hello`:

```
$ fig deploy foo/hello world
```

all instances of `{{ args[0] }}` would be replaced with `world`:

```
COMMAND: say hello *************************************************************
hello, world
```

> :bulb: Deployment arguments would be zero-indexed

### Variables

Argument substitution should be performed on variable values:

```yaml
# hello
---

- vars:
      who: "{{ args[0] }}"

- name: say hello
  command: echo "hello, {{ who }}"
```

### Missing Arguments

If the example profile `foo/hello` references a non-existent deployment argument:

```yaml
# hello
---

- name: say hello
  command: echo "hello, {{ args[0] }}"
```

```
$ fig deploy foo/hello
```

should deployment:

- **Succeed**, replacing all missing arguments with an empty string — i.e., run `echo "hello, "`
- **Succeed** without replacement — i.e., run `echo "hello, {{ args[0] }}"`
- **Fail** — ex., `Missing required deployment argument`

In other words, are deployment arguments optional or required?

## Notes

In practice, this is an unnamed counterpart to the "named arguments" approach of #4
